HealthcareApp/src/Healthcare.Infrastructure/Repository/ScheduleRepository.cs

```diff
@@ -0,0 +1,79 @@
+using Application.Abstractions;
+using Domain.Entities;
+using Healthcare.Application.DTOs.Schedule;
+using Healthcare.Infrastructure.Persistance;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Repository;
+
+public class ScheduleRepository : IScheduleRepository
+{
+    private readonly AppDbContext _appDbContext;
+
+    public ScheduleRepository(AppDbContext appDbContext) => _appDbContext = appDbContext;
+
+    public async Task<List<Schedule>> GetDoctorSchedule(string doctorId, int pageSize)
+        => await _appDbContext.Schedules.Where(s => s.DoctorId == doctorId).Take(pageSize).ToListAsync();
+
+    public async Task CreateDefaultWorkingScheduleAsync(string doctorId)
+    {
+        var workingHours = new List<Schedule>();
+        var startTime = new TimeSpan(8, 0, 0);
+        var endTime = new TimeSpan(16, 0, 0);
+
+        for (int i = 0; i < 5; i++)
+        {
+            var date = DateTime.Today.AddDays(i - (int)DateTime.Today.DayOfWeek + (int)DayOfWeek.Monday);
+            workingHours.Add(new Schedule
+            {
+                DoctorId = doctorId,
+                DayOfWeek = date.DayOfWeek,
+                Date = date,
+                StartTime = startTime,
+                EndTime = endTime
+            });
+        }
```

Comment:
You seem to add your business logic to the repository implementation. How about moving it to application and then passing the working schedule for this doctor as parameter? This will follow the clean architecture rules better